src/robotFactory.js

```diff
@@ -1,10 +1,74 @@
 'use strict';
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
+    this.coords = coords;
 
-class BaseRobot {}
+    if (this.coords.x === undefined) {
+      this.coords.x = 0;
+    }
 
-class FlyingRobot {}
+    if (this.coords.y === undefined) {
+      this.coords.y = 0;
+    }
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  };
+  goRight(step = 1) {
+    this.coords.x += step;
+  };
+  goBack(step = 1) {
+    this.coords.y -= step;
+  };
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  };
+
+  getInfo() {
+    // eslint-disable-next-line max-len
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    if (this.coords.z === undefined) {
+      this.coords.z = 0;
+    }
+  }
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords, chipVersion, maxLoadWeight, currentLoad) {
+    super(name, weight, coords, chipVersion);
+    this.maxLoadWeight = maxLoadWeight;
+
+    this.currentLoad = currentLoad === null ? null : { ...currentLoad };
```

Comment:
```suggestion
    this.currentLoad = currentLoad;
```
разве необходима проверка на null?